lib/puppet/provider/image_streamer_artifact_bundle/image_streamer.rb

```diff
@@ -33,13 +33,18 @@ def exists?
   def create
     current_resource = @resourcetype.find_by(@client, unique_id).first
     if current_resource
-      current_resource.update_name(@data['new_name']) if @data['new_name']
+      return true unless @data['new_name']
+      current_resource.update_name(@data['new_name'])
     elsif @data['artifact_bundle_path']
-      @resourcetype.create_from_file(@client, @data['artifact_bundle_path'], @data['name'])
+      create_from_file
     else
-      @resourcetype.new(@client, @data).create
+      @resourcetype.new(@client, @data).create.data
     end
-    true
+  end
+
+  def create_from_file
```

Comment:
No need for this method, these two lines can be left inside the `elsif @data['artifact_bundle_path']`, doesn't seem to change much.